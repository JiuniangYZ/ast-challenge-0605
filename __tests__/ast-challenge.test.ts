import createAst, { Params } from '../src/index'
import generate from '@babel/generator';

const expectCode = (ast) => {
  expect(
    generate(ast).code
  ).toMatchSnapshot();
}

it('code-challenge', () => {
  const ast = createAst({
    queryInterface: 'useCustomQuery',
    hookName: 'customHook',
    requestType: 'customRequest',
    responseType: 'customResponse',
    queryServiceMethodName: 'customService.customMethod()',
    keyName: 'customQuery'
  })
  expectCode(ast)
})
