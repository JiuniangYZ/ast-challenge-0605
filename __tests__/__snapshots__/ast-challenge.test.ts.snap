// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`code-challenge 1`] = `
"export interface useCustomQuery<TData> extends ReactQueryParams<customResponse, TData> {
  request?: customRequest;
}

const customHook = <TData = customResponse,>({
  request,
  options
}: useCustomQuery<TData>) => {
  return useQuery<customResponse, Error, TData>([\\"customQuery\\", request], () => {
    if (!customService) throw new Error(\\"Query Service not initialized\\");
    return customService.customMethod(request);
  }, options);
};"
`;
